### DISCUSSION TITLE
Selecting hail at surface from HRRR model

### DISCUSSION THREAD

**Discussion by btickell** [blaylockbk/Herbie#241] [Q&A]
I am trying to select specifically hail diameters at the surface.

```python
H = Herbie(
    "2021-07-19",
    model="hrrr",
    product="sfc",
    fxx=0
)

x = H.xarray()
```

The above code does work but I'd like to select just the hail field for performance / download speed reasons.

I tried using `:hail:` when calling xarray but no dice. I admit I am not very familiar with the underlying GRIB structure. Is selecting subfields  like this supported by Herbie?

**Comment by blaylockbk** [comment]
Good question.

Here is a table of some grib variable names...
https://www.nco.ncep.noaa.gov/pmb/docs/grib2/grib2_doc/grib2_table4-2-0-1.shtml

I'll have to sit at my computer to see what is actually in the HRRR sfc output, but I'd start to look for EFRHAIL which is effective radius of hail.

If you beat me to it, let me know what you find.

**Comment by btickell** [comment]
Ah thanks! Trying both EFRHAIL and HAIL (caps being important here) I do get results for `HAIL` but not for the effective hail.

I recieve an array with a `step` index of size 3 (all with value 0), but seems like only the first step has meaningful data. Do you have any insight into what this step refers to?

**Comment by blaylockbk** [comment]
It looks like there are three different levels the HAIL variable is output for.

```python
from herbie import Herbie
H = Herbie("2023-10-01", fxx=6)
H.inventory("HAIL")
```
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>grib_message</th>
      <th>start_byte</th>
      <th>end_byte</th>
      <th>range</th>
      <th>reference_time</th>
      <th>valid_time</th>
      <th>variable</th>
      <th>level</th>
      <th>forecast_time</th>
      <th>search_this</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>52</th>
      <td>53</td>
      <td>36161743</td>
      <td>36429492</td>
      <td>36161743-36429492</td>
      <td>2023-10-01</td>
      <td>2023-10-01 06:00:00</td>
      <td>HAIL</td>
      <td>entire atmosphere</td>
      <td>5-6 hour max fcst</td>
      <td>:HAIL:entire atmosphere:5-6 hour max fcst</td>
    </tr>
    <tr>
      <th>53</th>
      <td>54</td>
      <td>36429492</td>
      <td>36466353</td>
      <td>36429492-36466353</td>
      <td>2023-10-01</td>
      <td>2023-10-01 06:00:00</td>
      <td>HAIL</td>
      <td>0.1 sigma level</td>
      <td>5-6 hour max fcst</td>
      <td>:HAIL:0.1 sigma level:5-6 hour max fcst</td>
    </tr>
    <tr>
      <th>54</th>
      <td>55</td>
      <td>36466353</td>
      <td>36472988</td>
      <td>36466353-36472988</td>
      <td>2023-10-01</td>
      <td>2023-10-01 06:00:00</td>
      <td>HAIL</td>
      <td>surface</td>
      <td>5-6 hour max fcst</td>
      <td>:HAIL:surface:5-6 hour max fcst</td>
    </tr>
  </tbody>
</table>

I think you are looking for the surface parameter, so filter by that

```python
ds = H.xarray("HAIL:surface")
ds.hail.plot()
```
![image](https://github.com/blaylockbk/Herbie/assets/6249613/7b3b7e84-ad2b-450a-a562-617868ee86eb)



Now, I'm not too certain about the units. It might be worthwhile contacting the HRRR developers and ask them to clarify what this field represents.
https://github.com/NOAA-GSL/GSL-Regional-Model-Forum/discussions/
